hedera-mirror-web3/src/test/java/com/hedera/mirror/web3/service/AbstractContractCallServiceTest.java

```diff
@@ -101,4 +114,40 @@ protected ContractExecutionParameters getContractExecutionParameters(
                 .value(0L)
                 .build();
     }
+
+    protected void testEstimateGas(final RemoteFunctionCall<TransactionReceipt> functionCall, final Contract contract)
+            throws Exception {
+        testWeb3jService.setEstimateGas(true);
+
+        functionCall.send();
+        final var estimateGasUsedResult = longValueOf.applyAsLong(testWeb3jService.getTransactionResult());
+
+        final var actualGasUsed = gasUsedAfterExecution(getContractExecutionParameters(functionCall, contract));
+
+        // Then
+        assertThat(isWithinExpectedGasRange(estimateGasUsedResult, actualGasUsed))
+                .withFailMessage(ESTIMATE_GAS_ERROR_MESSAGE, estimateGasUsedResult, actualGasUsed)
+                .isTrue();
+
+        testWeb3jService.setEstimateGas(false);
+    }
+
+    public enum KeyType {
+        ADMIN_KEY(BigInteger.valueOf(1)),
```

Comment:
Clearer if you convert from integer to BigInteger in constructor.

```suggestion
        ADMIN_KEY(1),
```